### PR TITLE
Update Redis configuration to use new service

### PR DESCRIPTION
**Pull request by ccostino**
This changeset modifies our Redis configuration to use the new ElastiCache service (AWS Redis).

## Changes proposed in this pull request:
- Updates service binding to the new Redis service
- Updates the Redis config to enable TLS support for secure connections
- Updates the Redis config based on [recent redis-py updates](https://github.com/andymccurdy/redis-py#ssl-connections)

## security considerations:
- Makes uaa-extras more secure by using TLS connections with Redis
- Updates our internal service use to more modern implementations

### CODE REVIEW

**Comment by bengerman13** [comment]
Do we have a rollout plan to prevent invite/reset errors?

**Comment by ccostino** [comment]
We'll test the rollout of this in each environment before propagating it further; we ought to be able to rollback if there are any issues with this.  Are you thinking we need more than that, @bengerman13?  There's not much else to this, as long as the app can connect to the service we should be fine, and the services are all created in each environment according to @ChrisMcGowan.

**Comment by bengerman13** [comment]
Yeah - the issue is that redis stores the redemption codes for password resets and invites. If someone gets an invite before we cut over and tries to redeem it after, they'll get an error. Same thing with password resets, although there's probably a shorter time in flight for those.

I think the invite problem the other day highlights how many of these we might have in-flight at any given time

**Comment by bengerman13** [comment]
Also, most other services create their required services in their pipeline - we should probably do the same here. I know this pipeline doesn't currently, but I think it's relevant enough to this update to add it.

**Comment by ccostino** [comment]
Regarding your first point, I don't think there's anything we can really do about that except for posting a StatusPage notice and doing this at an off-hours time.  We don't have a backup system that we can cut-over to while performing the upgrade, we just have to do it.  If possible, we could push a quick update first that just shuts off access, takes away the links, and puts a maintenance message up first, then perform these updates.  That seems like the best way to handle this to me.

For the second part regarding the services, we can take a look to see what else we'd need to do to update the pipeline.

**Comment by bengerman13** [comment]
We could have temporary config that writes to the new redis and reads from the new and old (ignoring errors if the old is unavailable)

I don't think statuspage is enough in this case, since invites are for prospective users, who probably aren't subscribed to or aware of our alerts

**Comment by ccostino** [comment]
Hmm, that could work...  prefer reading from the new if the record is in both, otherwise fall back.  Definitely a phased approach.  What this is telling me is we need some way of having a maintenance mode for this. :-)

You're right that a StatusPage update isn't going to do much for prospective new users trying things out, hence my suggestion for replacing the signup page with one that has the links shut off/removed and a maintenance message instead.

**Comment by bengerman13** [comment]
If we go with the messaging, I'd suggest we:
- put up messaging announcing this for at least 2 weekdays
- turn off invite functionality
- wait until there are zero invite keys in Redis
- update app, enable invites, remove warning

To me, it seems easier to support two redis connections at the same time, but I could be wrong.

(We could probably also just dump the contents of the old redis into the new redis, manually, but I'd worry a little bit about getting that wrong and upsetting users)

File: uaaextras/templates/maintenance.html [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{% extends "_layout.html" %}
+
+{% block header %}System Maintenance{% endblock %}
+
+{% block content %}
+<div class="island-content">
+    <p>Our system is currently undergoing scheduled maintenance and will be back online soon.</p>
```
Comment (bengerman13) [inline uaaextras/templates/maintenance.html:7]: of the options discussed, this really sounds like the hardest and worst experience for users to me

Comment (ccostino) [reply uaaextras/templates/maintenance.html:7]: Agreed after our discussion yesterday, but this might be useful to have just in case for the future.

**Comment by ccostino** [comment]
This is now ready to go, paired with https://github.com/cloud-gov/cg-uaa-extras-redis-migrator - we'd push this migrator app manually in production first, then we can merge this PR and let it deploy all of the way through.  Once deployment is done and we've confirmed things are working properly, we just delete the migrator app and then can unbind and delete the old Redis service.

**Review by bengerman13 — APPROVED** [review]

**Pull request merged**